src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactoryImpl.java

```diff
@@ -1,19 +1,35 @@
 package com.atlassian.bitbucket.jenkins.internal.client;
 
-import com.atlassian.bitbucket.jenkins.internal.client.supply.BitbucketCapabilitiesSupplier;
 import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.model.AtlassianServerCapabilities;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import okhttp3.HttpUrl;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.atlassian.bitbucket.jenkins.internal.util.SystemPropertiesConstants.CAPABILITIES_CACHE_DURATION_KEY;
+import static com.atlassian.bitbucket.jenkins.internal.util.SystemPropertyUtils.parsePositiveLongFromSystemProperty;
+
 public class BitbucketClientFactoryImpl implements BitbucketClientFactory {
 
+    /**
+     * Cache duration for the capabilities response. Defaults to 1 hour in ms.
+     */
+    public static final long CAPABILITIES_CACHE_DURATION =
+            parsePositiveLongFromSystemProperty(CAPABILITIES_CACHE_DURATION_KEY, 360000);
+    private final Cache<HttpUrl, AtlassianServerCapabilities> capabilitiesCache = CacheBuilder.newBuilder()
+            .expireAfterAccess(CAPABILITIES_CACHE_DURATION, TimeUnit.MILLISECONDS)
```

Comment:
maybe `expireAfterWrite` is better? This way the cache effectively never expire if we get one cache hit at least once an hour.